inc/saiswitch.h

```diff
@@ -1827,6 +1880,195 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_NAT_ENABLE,
 
+    /**
+     * @brief SDK/Driver can access device through file system support.
+     *
+     * TRUE - Device access by file system.
+     * FALSE - Device access not supported directly by driver.
+     * Platform adaption device read/write API should be provided by the Host Adapter.
+     * @type bool
+     * @flags READ_ONLY
+     */
+    SAI_SWITCH_ATTR_HARDWARE_ACCESS_SUPPORT_BY_SYSFS,
+
+    /**
+     * @brief Switch hardware access bus MDIO/I2C/CPLD
+     *
+     * @type sai_switch_hardware_access_bus_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @default SAI_SWITCH_HARDWARE_ACCESS_BUS_MDIO
+     * @condition SAI_SWITCH_ATTR_HARDWARE_ACCESS_SUPPORT_BY_SYSFS == false
+     */
+    SAI_SWITCH_ATTR_HARDWARE_ACCESS_BUS,
+
+    /**
+     * @brief Platform context information
+     *
+     * Platform context information provided by the host adaptor to driver.
+     * This information is Host adaptor specific, typically used for maintain
+     * synchronization and device information. Driver will give this context back
+     * to adaptor as part of call back sai_switch_register_read/write_fn API.
+     *
+     * Use void as pointer.
+     *
+     * @type sai_pointer_t void
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
```

Comment:
why this is mandatory on created ? seems like platform context should be something that should be obtained from switch which was created after switch creation

Reply:
PHY driver will call back API to access registers to NOS. NOS will take care of reaching PHY, During this time we need protection(locking) for bus access. PHY register access will happen during the switch_create also. So this it opaque date to PHY driver, NOS will provide during switch create and NOS will get back from PHY driver. Data inside this specific to NOS implementation. Like other  notification call back API's, pointer can be exchanged between NOS and PHY driver.

Reply:
not sure if i understand this fully, switch ID should be enough to identify registers for expected switch, internally SAI can associate any platform context with returned switch ID. there should be no callback for calling such APIs like this one. Those should be synchronous APIs and any necessary locking should happen inside internal vendor SAI implementation. Other vendors may not have concept of "context" etc. If you really want callback for reading registers (which i should advise to not go this path) then declare notification for switch like other notifications .
IMO this should be synchronous API

Reply:
@lguohan Can can take a look at this? Since concept of registers would be per platform and it seems like it can't be stored as attributes, then some special cases and database would need to be added in SONIC if some of registers would be modified

Reply:
@AshokDaparthi , syncd with tian, can you check the type to sai_uint64_t?